### PR TITLE
Rearrange main executables and tests

### PR DESCRIPTION
**Pull request by mseri**
This moves the executables and tests in different scopes, reorganising and clarifying the dependencies.
This is a first step, further improvements will be to further refine the dependencies among those folders and replace `stdext` with the appropriate `xapi-stdext-` submodules.

### CODE REVIEW

File: ocaml/xapi/jbuild [hunk 1]
```diff
@@ -67,11 +67,11 @@ let () = Printf.ksprintf Jbuild_plugin.V1.send {|
...
- ((names (xapi_main suite suite_alcotest quicktest))
-  (public_names (xapi - - quicktestbin))
-  (package xapi)
+(library
+ ((name xapi_internal)
+  (wrapped false)
   (flags (:standard -bin-annot %s -warn-error +a-3-4-6-9-27-28-29-52))
+  (modules (:standard \ xapi_main quicktest))
   (libraries (
    alcotest
```
Comment (gaborigloi) [inline ocaml/xapi/jbuild]: Can we also remove this?

**Comment by gaborigloi** [comment]
Will it be easier to merge https://github.com/xapi-project/xen-api/pull/3468 first and then update this PR to move the new / modified files to their correct place?
Or I guess we should merge this first if reviewing those PRs would take long , and then  https://github.com/xapi-project/xen-api/pull/3468 will have to be updated.

**Comment by mseri** [comment]
Yes, it will not be a problem to rebase after your changes in #3648. I think that should go first, once it passes travis tests

**Comment by mseri** [comment]
Rebased on top of #3468.

`ls ocaml/xapi` now fits in a full-screen terminal :tada: :fearful:

File: ocaml/tests/jbuild [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+(alias
+((name runtest)
+ (package xapi)
+(deps (suite_alcotest.exe (files_recursively_in test_data)))
+(action (run ${<}))
+)
+)
+|} (flags rewriters) coverage_rewriter
```
Comment (robhoes) [inline ocaml/tests/jbuild]: Do we still need to do this?

Comment (gaborigloi) [reply ocaml/tests/jbuild]: Only for coverage, as far as I know.

Comment (gaborigloi) [reply ocaml/tests/jbuild]: So I think we could remove `flags`.

Comment (mseri) [reply ocaml/tests/jbuild]: It is still needed until we port `rpc` to `jbuilder`. A port is ready but there are some issues with `camlp4` and the way it interacts with ppx in `xapi-idl`, so I'd rather wait until the port is complete

**Comment by mseri** [comment]
I've rebased to avoid the conflicts. I am going to merge this if Travis is happy

**Comment by mseri** [comment]
The xenserver-build-env failure is due to idl changes not yet propagated. The Travis build with xs-opam has successfully passed. Merging

**Pull request merged**